http/sys_rekey_test.go

```diff
@@ -177,24 +177,32 @@ func TestSysRekey_Update(t *testing.T) {
 	testResponseStatus(t, resp, 200)
 	testResponseBody(t, resp, &rekeyStatus)
 
-	resp = testHttpPut(t, token, addr+"/v1/sys/rekey/update", map[string]interface{}{
-		"nonce": rekeyStatus["nonce"].(string),
-		"key":   hex.EncodeToString(master),
-	})
-
 	var actual map[string]interface{}
-	expected := map[string]interface{}{
-		"complete":         true,
-		"nonce":            rekeyStatus["nonce"].(string),
-		"backup":           false,
-		"pgp_fingerprints": interface{}(nil),
-	}
-	testResponseStatus(t, resp, 200)
-	testResponseBody(t, resp, &actual)
-
-	keys := actual["keys"].([]interface{})
-	if len(keys) != 5 {
-		t.Fatalf("bad: %#v", keys)
+	var expected map[string]interface{}
+
+	for i, key := range keys {
+		resp = testHttpPut(t, token, addr+"/v1/sys/rekey/update", map[string]interface{}{
+			"nonce": rekeyStatus["nonce"].(string),
+			"key":   hex.EncodeToString(key),
+		})
+
+		actual = map[string]interface{}{}
+		expected = map[string]interface{}{
+			"complete":         false,
+			"nonce":            rekeyStatus["nonce"].(string),
+			"backup":           false,
+			"pgp_fingerprints": interface{}(nil),
+		}
+		if i+1 == len(keys) {
+			expected["complete"] = true
+		}
+		testResponseStatus(t, resp, 200)
+		testResponseBody(t, resp, &actual)
+	}
```

Comment:
Should each loop compare actual to expected? Below the test is checking `reflect.DeepEqual(actual, expected)`